### PR TITLE
Add require check

### PR DESCRIPTION
**Pull request by fntz**
Hello,

fix for https://github.com/scapegoat-scala/scapegoat/issues/730

scala2/scala3

### CODE REVIEW

File: src/test/scala-2/com/sksamuel/scapegoat/inspections/AvoidRequireTest.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          class Test(val x: Int) {
+            require(x == 1, "oops")
+          }
+          """.stripMargin
+
+      compileCodeSnippet(code)
+      compiler.scapegoat.feedback.warnings.size shouldBe 1
+    }
+
+    "should not return warnin on own require method" in {
```
Comment (saeltz) [inline src/test/scala-2/com/sksamuel/scapegoat/inspections/AvoidRequireTest.scala:36]:
```suggestion
    "should not return warning on own require method" in {
```

File: src/test/scala-3/com/sksamuel/scapegoat/inspections/AvoidRequireTest.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          class Test(val x: Int) {
+            require(x == 1, "oops")
+          }
+          """.stripMargin
+
+      val feedback = runner.compileCodeSnippet(code)
+      feedback.warnings.assertable.size shouldBe 1
+    }
+
+    "should not return warnin on own require method" in {
```
Comment (saeltz) [inline src/test/scala-3/com/sksamuel/scapegoat/inspections/AvoidRequireTest.scala:35]:
```suggestion
    "should not return warning on own require method" in {
```

File: src/main/scala-3/com/sksamuel/scapegoat/inspections/AvoidRequire.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    explanation = "Using require throws an untyped Exception."
+  ) {
+
+  import tpd.*
+
+  def inspect(feedback: Feedback[SourcePosition], tree: tpd.Tree)(using ctx: Context): Unit = {
+    val traverser = new InspectionTraverser {
+      def traverse(tree: Tree)(using Context): Unit = {
+        tree match {
+          case f@Apply(ident: Ident, _) if ident.name.toString == "require" =>
```
Comment (saeltz) [inline src/main/scala-3/com/sksamuel/scapegoat/inspections/AvoidRequire.scala:25]: It seems `f` is unused, isn't it?
```suggestion
          case Apply(ident: Ident, _) if ident.name.toString == "require" =>
```

Comment (fntz) [reply src/main/scala-3/com/sksamuel/scapegoat/inspections/AvoidRequire.scala]: right. Removed unused `f`

**Comment by saeltz** [comment]
@mccartney, would you issue a new release?

**Comment by mccartney** [comment]
Thanks for another contribution!

**Pull request merged**